jetty-xml/src/main/java/org/eclipse/jetty/xml/XmlConfiguration.java

```diff
@@ -1777,40 +1646,168 @@ public Object get(String elementName, boolean mandatory) throws Exception
+                {
+                    _arguments.add(value(obj, child));
+                    _names.add(child.getAttribute("name"));
+                }
+            }
+
+            private NamedArgs(List<Object> arguments, List<String> names)
+            {
+                _class = null;
+                _arguments = arguments;
+                _names = names;
+            }
+
+            Object[] applyTo(Executable executable)
+            {
+                Object[] args = matchArgsToParameters(executable);
+                if (args == null && _class != null)
                 {
-                    for (String a : StringUtil.csvSplit(null, attr, 0, attr.length()))
+                    // Could this be an empty varargs match?
+                    int count = executable.getParameterCount();
+                    if (count > 0 && executable.getParameterTypes()[count - 1].isArray())
                     {
-                        // create a fake node
-                        XmlParser.Node n = new XmlParser.Node(null, elementName, null);
-                        n.add(a);
-                        values.add(n);
+                        try
+                        {
+                            // Does a non varargs method exist?
+                            Class<?>[] types = Arrays.copyOf(executable.getParameterTypes(), count - 1);
+                            if (executable instanceof Constructor)
+                                _class.getConstructor(types);
+                            else
+                                _class.getMethod(((Method)executable).getName(), types);
+                        }
+                        catch (NoSuchMethodException e)
+                        {
+                            // There is not a no varArgs alternative so let's try a null varArgs match
+                            args = asEmptyVarArgs(executable.getParameterTypes()[count - 1]).matchArgsToParameters(executable);
+                        }
+                        catch (Exception e)
+                        {
+                            LOG.ignore(e);
+                        }
                     }
                 }
+                return args;
+            }
 
-                for (int i = 0; i < _next; i++)
+            NamedArgs asEmptyVarArgs(Class<?> varArgType)
+            {
+                List<Object> arguments = new ArrayList<>(_arguments);
+                arguments.add(Array.newInstance(varArgType.getComponentType(), 0));
+                List<String> names = new ArrayList<>(_names);
+                names.add(null);
+                return new NamedArgs(arguments, names);
+            }
+
+            Object[] matchArgsToParameters(Executable executable)
+            {
+                int count = executable.getParameterCount();
+
+                // No match of wrong number of parameters
+                if (count != _arguments.size())
+                    return null;
```

Comment:
This does not cover the case of `executable(String first, String... others)` with `args=["one", "two", "three"]`, where `count==2` but `_arguments.size()==3`.
Not sure it was covered before, but while we are at it...

Reply:
I don't think we can support that, at least not easily.  To call `executable(String first, String... others)` with `args=["one", "two", "three"]` you will need to do:
```xml
   <Call name=executable>
     <Arg>one</Arg>
     <Arg>
       <Array type=String>
         <Item>two</Item>
         <Item>three</Item>
       </Array>
     </Arg>
   </Call>
```